app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/map/BeaconMapBackground.java

```diff
@@ -0,0 +1,214 @@
+package com.nexenio.bleindoorpositioningdemo.ui.beaconview.map;
+
+import android.graphics.Bitmap;
+import android.graphics.Point;
+import android.support.annotation.NonNull;
+
+import com.nexenio.bleindoorpositioning.location.Location;
+import com.nexenio.bleindoorpositioning.location.projection.CanvasProjection;
+
+public class BeaconMapBackground {
+
+    private Bitmap imageBitmap;
+
+    private double metersPerPixel;
+
+    private double bearing;
+
+    private Location topLeftLocation;
+    private Location bottomRightLocation;
+
+    private Point topLeftPoint;
+    private Point bottomRightPoint;
+
+    private CanvasProjection projection = new CanvasProjection();
+
+    private BeaconMapBackground(Bitmap imageBitmap) {
+        this.imageBitmap = imageBitmap;
+        projection.setCanvasWidth(imageBitmap.getWidth());
+        projection.setCanvasHeight(imageBitmap.getHeight());
+        topLeftPoint = new Point(0, 0);
+        bottomRightPoint = new Point(imageBitmap.getWidth(), imageBitmap.getHeight());
+    }
+
+    public Location getLocation(Point point) {
+        return getLocation(point, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public Point getPoint(Location location) {
+        return getPoint(location, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public static float getMetersPerPixel(Location firstReferenceLocation, Point firstReferencePoint, Location secondReferenceLocation, Point secondReferencePoint) {
+        double distanceInPixels = getPixelDistance(firstReferencePoint, secondReferencePoint);
+        double distanceInMeters = firstReferenceLocation.getDistanceTo(secondReferenceLocation);
+        return (float) (distanceInMeters / distanceInPixels);
```

Comment:
Can throw an `ArithmeticException` for division by 0 as `distanceInPixels` can be 0.

Reply:
Fixed